components/esp_http_client/esp_http_client.c

```diff
@@ -136,7 +136,7 @@ static esp_err_t _clear_connection_info(esp_http_client_handle_t client);
 #define ASYNC_TRANS_CONNECTING 0
 #define ASYNC_TRANS_CONNECT_PASS 1
 
-static const char *DEFAULT_HTTP_USER_AGENT = "ESP32 HTTP Client/1.0";
+static const char *DEFAULT_HTTP_USER_AGENT = CONFIG_ESP_HTTP_CLIENT_USER_AGENT;
```

Comment:
How about we add an additional member to `esp_http_client_config_t` as `const char *user_agent`?

And then during initialization we can use something like:

```
const char *user_agent = config->user_agent == NULL ? DEFAULT_HTTP_USER_AGENT : config->user_agent;
esp_http_client_set_header(client, "User-Agent", user_agent);
```
```

Reply:
Sounds good to me. :)